hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFileReader.java

```diff
@@ -56,7 +56,6 @@
   private static final byte[] oldMagicBuffer = new byte[4];
   private static final byte[] magicBuffer = new byte[6];
   private final Schema readerSchema;
-  private HoodieLogBlock nextBlock = null;
```

Comment:
does this cause additional memory overhead?

Reply:
For this issue. The following is my analysis. Essentially, we pass an inputstream from HoodieLogFileReader to the LogBlocks. An inputstream can be shared in more than 1 logblock. A list of logBlocks are kept until they are merged. So until all logblocks holding the inputstream generated in HoodieLogFileReader are GC'd, they will continue to hold memory.

Reply:
Per the screenshot in the ticket,  most retained memory comes from `private List<IndexedRecord> records;` in the AvroDataBlock and not the inputStream itself? The inputstream being held does not seem to have causing these issues..  
This might be just a large data block being loaded into memory?